### PR TITLE
Now we are able to catch NetworkMigrationExceptions

### PR DESCRIPTION
**Pull request by dvygolov**

### CODE REVIEW

File: TLSharp.Core/Network/MtProtoSender.cs [hunk 1]
```diff
@@ -305,6 +305,12 @@ private bool HandleRpcResult(ulong messageId, int sequence, BinaryReader message
                     var dcIdx = int.Parse(resultString);
                     throw new UserMigrationException(dcIdx);
                 }
+		else if (errorMessage.StartsWith("NETWORK_MIGRATE_"))
```
Comment (knocte) [inline TLSharp.Core/Network/MtProtoSender.cs]: missing one space in this line (indentation incorrect)

File: TLSharp.Core/TelegramClient.cs [hunk 2]
```diff
@@ -1,170 +1,170 @@
...
-using TeleSharp.TL.Auth;
-using TeleSharp.TL.Contacts;
-using TeleSharp.TL.Help;
-using TeleSharp.TL.Messages;
-using TeleSharp.TL.Upload;
-using TLSharp.Core.Auth;
-using TLSharp.Core.MTProto.Crypto;
-using TLSharp.Core.Network;
-using TLSharp.Core.Utils;
-using TLAuthorization = TeleSharp.TL.Auth.TLAuthorization;
```
Comment (knocte) [inline TLSharp.Core/TelegramClient.cs]: why are you removing all these lines and adding them back? it seems there's some discrepancy with line endings

File: TLSharp.Core/Network/MtProtoSender.cs (hunk 1 above)
Comment (dvygolov) [reply TLSharp.Core/Network/MtProtoSender.cs]: Hmm, can't see it in my original file( Looks like a github bug.
image

Comment (knocte) [reply TLSharp.Core/Network/MtProtoSender.cs]: haha, no, there's no github bug here, see it in raw: https://github.com/sochix/TLSharp/commit/a2322b6ce433b21ed648941683d96ed0caaa5ab7.patch (you may have a misconfigured editor that hides you differences between tabs and spaces)

**Comment by dvygolov** [comment]
Ok, corrected formatting typos

**Comment by knocte** [comment]
Looks better! but can you now squash all commits into 1 to avoid messy commit history? (and to not have git-blame point to whitespace changes) You can do this via `git rebase -i HEAD~6` and then replacing all "pick" words of commits with "squash" (except the first commit).

**Comment by dvygolov** [comment]
Am I done now?) That's my first pull request ever, sorry for messing things up)

**Comment by knocte** [comment]
Look at the commits tab of the pull requests: you can still see 8 commits instead of 1. Did you do what I said to squash the commits?

**Comment by dvygolov** [comment]
Definitely I did that. Can't you just squash all these commits, when you merge the pull request to the master branch? As I can see, [there is such a func in GitHub](https://github.com/blog/2141-squash-your-commits).

**Comment by dvygolov** [comment]
@knocte, I think I managed to squash it now, merge, please.

**Comment by knocte** [comment]
Great! only 1 nitpick, can you include in the commit message the text `Fixes #568` so that we link to the issue you created?

**Comment by dvygolov** [comment]
@knocte, done.

**Comment by knocte** [comment]
Good work!

**Pull request merged**